### PR TITLE
NET-272: Data query endpoints fetch directly from storage

### PR DESCRIPTION
**Pull request by teogeb**
Replaced `networkNode.requestResendFrom`,  `networkNode.requestResendRange` and  `networkNode.requestResendLast`, calls with equivalent Storage calls.

### CODE REVIEW

File: src/http/DataMetadataEndpoints.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
 import express, { Request, Response } from 'express'
+import { Storage } from '../storage/Storage'
+
+const parseIntIfExists = (x: string) => {
```
Comment (harbu) [inline src/http/DataMetadataEndpoints.ts:4]: should the type of x be then `string | undefined`?

File: src/http/index.ts [hunk 2]
```diff
@@ -25,11 +25,11 @@ export const start = (
     app.use(cors())
 
     // Rest endpoints
-    app.use('/api/v1', dataQueryEndpoints(networkNode, streamFetcher, metricsContext))
+    app.use('/api/v1', dataQueryEndpoints(cassandraStorage!, streamFetcher, metricsContext))
```
Comment (harbu) [inline src/http/index.ts:28]: I'm assuming these `cassandraStorage!` statements will be covered by an `if`-statement later on? I.e. `if (cassandraStorage !== undefined)`

**Review by harbu — APPROVED** [review]
LGTM

File: src/http/DataMetadataEndpoints.ts (hunk 1 above)
Comment (teogeb) [reply src/http/DataMetadataEndpoints.ts]: Fixed

File: src/http/index.ts (hunk 2 above)
Comment (teogeb) [reply src/http/index.ts:28]: Would it make sense to put that validation to config level? We'd check that if isStorageNode is true, there must be all required configurations for storage node (`cassandraStorage`, `storageConfig` etc.). And we could make a type definition `BrokerConfig|StorageNodeConfig`.

Comment (harbu) [reply src/http/index.ts:28]: Makes sense, a brings us one step closer to a potential separation between broker / storage.

Comment (teogeb) [reply src/http/index.ts:28]: Implemented here: https://github.com/streamr-dev/broker/pull/254

**Pull request merged**